packages/sign_in_with_apple/ios/Classes/SwiftSignInWithApplePlugin.swift

```diff
@@ -0,0 +1,199 @@
+                            details: nil // call -> call might have lead to `Unsupported value: <FlutterMethodCall: 0x6000000a2640> of type FlutterMethodCall` error
+                        )
+                    )
+
+                    return
+                }
+
+                let appleIDProvider = ASAuthorizationAppleIDProvider()
+                appleIDProvider.getCredentialState(forUserID: userIdentifier) { credentialState, error in
+                    if let error = error {
+                        result(
+                            FlutterError(
+                                code: "ERR",
+                                message: "Failed to get credentials state: \(error)",
+                                details: nil
+                            )
+                        )
+
+                        return
+                    }
+
+                    switch credentialState {
+                    case .authorized:
+                        result("authorized")
+
+                    case .revoked:
+                        result("revoked")
+
+                    case .notFound:
+                        result("notFound")
+
+                    default:
+                        result(
+                            FlutterError(
+                                code: "ERR",
+                                message: "Unexpected credential state: \(credentialState)",
+                                details: nil
+                            )
+                        )
+                    }
+                }
+
+            default:
+                result(FlutterMethodNotImplemented)
+
+                return
+            }
+        } else {
+            result(
+                FlutterError(
+                    code: "ERR",
+                    message: "Unsupported iOS version",
+                    details: nil
+                )
+            )
+        }
+    }
+}
+
+@available(iOS 13.0, *)
+class AYSignInWithAppleAuthorizationController: NSObject, ASAuthorizationControllerDelegate {
+    enum CurrentAction {
+        case attemptWithPasswords
+        case attemptWithoutPasswords
+    }
+
+    var resultCallback: FlutterResult
+
+    var currentAction: CurrentAction = .attemptWithPasswords
+
+    init(_ callback: @escaping FlutterResult) {
+        resultCallback = callback
+    }
+
+    public func performRequests() {
+        let appleIDProvider = ASAuthorizationAppleIDProvider()
+        let request = appleIDProvider.createRequest()
+        request.requestedScopes = [.fullName, .email]
+
+        var requests: [ASAuthorizationRequest] = [request]
+
+        if currentAction == .attemptWithPasswords {
+            let passwordProvider = ASAuthorizationPasswordProvider()
+            let passwordRequest = passwordProvider.createRequest()
+
+            requests.append(passwordRequest)
+        }
+
+        let authorizationController = ASAuthorizationController(authorizationRequests: requests)
```

Comment:
So if one of the requests fails, the `didCompleteWithError` will be called?

And if we have a request with a password, we then try to request without it?

Reply:
Yup, that's exactly how this works.

Tried a bunch of attempts to get the correct error code, so we could fail early if appropriate, but couldn't figure it out, as the detailed error class is just internal.

Also seems broken from the API perspective, as Apple recommends to request both to not create a second account for users who already have one. But since there is no open bug tracker I just have that forums comment to work with (which explained the error, but didn't suggest any workaround / alternative approach).